### PR TITLE
Use POST for remediations playbook endpoint instead of GET

### PR DESCRIPTION
**Pull request by chris1984**
#### What are the changes introduced in this pull request?

* When the remediations playbook URL contains host UUIDs as query parameters (GET /remediations/:id/playbook?hosts=uuid1,uuid2,...), the URLRemediationsRetriever now extracts the hosts from the URL and sends them as a JSON array in the body of a POST request instead. This prevents HTTP 400 errors caused by URL length limits when many hosts (e.g. 321+) are included. URLs without a hosts query parameter continue to use GET with the original behavior unchanged.

#### Considerations taken when implementing this change?

- Backward compatibility: Only switches to POST when hosts are present in the URL. URLs without hosts still use GET, so existing callers (e.g. download_rh_playbook without host params) are unaffected.
- Other query parameters: Any non-hosts query parameters on the URL are preserved when stripping hosts.
- Minimal scope: The change is contained to URLRemediationsRetriever — no changes needed to the job template, controller, or cloud connector flow.

#### What are the testing steps for this pull request?
* Setup a non iop devel box or production install
* Import a manifest and sync some repos
* Register a host and configure some vulnerabilities
* Configure `cloud connector`
* Login to `console.redhat.com` and try to remediate a recommendation and watch the playbook connect to your Foreman server and run then send back up a request as a `POST` instead of `GET`

Screenshot of rh_cloud recommendations before:
Screenshot_20-4-2026_143140_ip-10-0-167-106 rhos-01 prod psi rdu2 redhat com

Screenshot of rh_cloud recommendations after:
Screenshot_20-4-2026_143156_ip-10-0-167-106 rhos-01 prod psi rdu2 redhat com

Screenshot of job:
Screenshot_20-4-2026_143128_ip-10-0-167-106 rhos-01 prod psi rdu2 redhat com

Screenshots of success on console.redhat.com and playbook run
Screenshot_20-4-2026_143216_console redhat com
Screenshot_20-4-2026_14314_console redhat com

#### Output of `POST` request:

```ruby
22:45:18 rails.1   | 2026-04-20T22:45:18 [D|tem|aaae8aac] Querying playbook via POST at: https://cert.cloud.redhat.com/api/remediations/v1/remediations/71feeb38-1a82-4132-9b36-ad8379556772/playbook with payload: ["61b8abb6-dcc6-43f8-b52a-990464c87341"] and headers: {:content_type=>:json}
22:45:18 rails.1   | 2026-04-20T22:45:18 [D|tem|aaae8aac] Response headers for request url https://cert.cloud.redhat.com/api/remediations/v1/remediations/71feeb38-1a82-4132-9b36-ad8379556772/playbook are: {:server=>"openresty", :content_type=>"text/vnd.yaml; charset=utf-8", :content_length=>"4853", :x_rh_insights_request_id=>"0c27275796b44a3db507d9e1ad7e4d07", :x_powered_by=>"Express", :etag=>"W/\"1284-TQie8Wa9G1TPQMaPKYkYs/ZXGWI\"", :content_disposition=>"attachment;filename=\"test4-1776739525501.yml\"", :x_content_type_options=>"nosniff", :date=>"Tue, 21 Apr 2026 02:45:25 GMT", :connection=>"keep-alive", :set_cookie=>["3ba1432eca9ab72ebcff858daecadbf5=de6d939ec411a941178f66fe946f4d8e; path=/; HttpOnly; Secure; SameSite=None"], :x_rh_edge_request_id=>"a1d6419f", :x_rh_edge_reference_id=>"0.9b6bdc17.1776739525.a1d6419f", :x_rh_edge_cache_status=>"NotCacheable from child", :x_frame_options=>"SAMEORIGIN", :strict_transport_security=>"max-age=31536000; includeSubDomains"}
22:45:18 rails.1   | 2026-04-20T22:45:18 [D|tem|aaae8aac] Got playbook response: ---
22:45:18 rails.1   |  aaae8aac | # Red Hat Insights has recommended one or more actions for you, a system administrator, to review and if you
22:45:18 rails.1   |  aaae8aac | # deem appropriate, deploy on your systems running Red Hat software. Based on the analysis, we have automatically
22:45:18 rails.1   |  aaae8aac | # generated an Ansible Playbook for you. Please review and test the recommended actions and the Playbook as
22:45:18 rails.1   |  aaae8aac | # they may contain configuration changes, updates, reboots and/or other changes to your systems. Red Hat is not
22:45:18 rails.1   |  aaae8aac | # responsible for any adverse outcomes related to these recommendations or Playbooks.
```

### CODE REVIEW

File: app/services/foreman_rh_cloud/url_remediations_retriever.rb [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
     def initialize(url:, organization_id:, payload: '', headers: {}, logger: Logger.new(IO::NULL))
       super(logger: logger)
 
-      @url = url
+      parsed_url = URI.parse(url)
+      query_params = parsed_url.query ? CGI.parse(parsed_url.query) : {}
+      hosts_param = query_params.delete('hosts')
+
+      if hosts_param.present?
+        @host_uuids = hosts_param.flat_map { |v| v.split(',') }
```
[the remark this answers is not in the document]
Comment (nofaralfasi) [reply app/services/foreman_rh_cloud/url_remediations_retriever.rb]: I agree we should address this comment. We can implement it like this:
```
if hosts_param.present?
  @host_uuids = hosts_param.flat_map { |v| v.split(',') }.map(&:strip).reject(&:blank?)
  @host_uuids = nil if @host_uuids.empty?
  # ... rest of the logic
end
```

**Review by nofaralfasi — COMMENTED** [review]
Nice fix for the URL length limit issue! The approach of moving hosts to the POST body is solid.
A couple of test suggestions to cover edge cases:
Empty hosts param - Verify that ?hosts= (or ?hosts=uuid-1,) falls back to GET instead of sending [''] as the payload. Consider adding .map(&:strip).reject(&:blank?) to the host extraction:
```
@host_uuids = hosts_param.flat_map { |v| v.split(',') }.map(&:strip).reject(&:blank?)
@host_uuids = nil if @host_uuids.empty?
```
Multiple hosts params - A test for `?hosts=uuid-1&hosts=uuid-2` would document that the flat_map correctly merges these into a single array.

**Comment by chris1984** [comment]
@nofaralfasi updated and added a test case like you wanted. I tested and was still able to remediate from `console.redhat.com` to my box and it show success both on Satellite and `console.redhat.com`

**Review by nofaralfasi — APPROVED** [review]
Thanks Chris, LGTM.
I’ll leave it to you to decide whether we can merge it now or if we should wait for QE ACK first.
